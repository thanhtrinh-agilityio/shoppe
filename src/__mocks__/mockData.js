export const PRODUCT_ITEM = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: 'Sold out',
}

export const PRODUCT_DETAIL = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: 'Sold out',
  quantity: 1,
  introduction:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. ',
}

export const LIST_PRODUCT = [
  {
    id: 1,
    imageUrl: '/images/liraEarring.svg',
    name: 'Lira Earrings',
    price: 20,
    status: '-25%',
  },
  {
    id: 2,
    imageUrl: '/images/liraEarring.svg',
    name: 'Lira Earrings',
    price: 20,
    status: '',
  },
  {
    id: 3,
    imageUrl: '/images/liraEarring.svg',
    name: 'Kaede Hair Pin Set Of 3',
    price: 20,
    status: '',
  },
  {
    id: 4,
    imageUrl: '/images/liraEarring.svg',
    name: 'Hair Pin Set of 3',
    price: 20,
  },
  {
    id: 5,
    imageUrl: '/images/liraEarring.svg',
    name: 'Plaine Necklace',
    price: 20,
    status: 'Sold out',
  },
  {
    id: 6,
    imageUrl: '/images/liraEarring.svg',
    name: 'Yuki Hair Pin Set of 3',
    price: 20,
    status: '',
  },
]
