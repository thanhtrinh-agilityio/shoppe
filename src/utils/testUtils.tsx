import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <CartProvider>{component}</CartProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
