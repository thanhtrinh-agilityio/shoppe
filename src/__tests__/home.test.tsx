import React from 'react'

// Pages
import Home from '@pages/products'

// Utils
import { render } from '@utils/testUtils'

// Mocks
import { LIST_PRODUCT } from '@mocks/mockData'

describe('Home render', () => {
  it('Should show match Home DOM Snapshot', async () => {
    const { container } = render(<Home products={LIST_PRODUCT} error="" />)

    expect(container).toMatchSnapshot()
  })
})
