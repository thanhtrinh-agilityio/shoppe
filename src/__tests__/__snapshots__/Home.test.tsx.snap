// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render Should show match Home DOM Snapshot 1`] = `
<div>
  <div
    class="css-butksl"
  >
    <h2
      class="chakra-heading css-aj5n6i"
    >
      Shop The Latest
    </h2>
    <div
      class="css-17dci3b"
    >
      <div
        class="css-1i5wex2"
      >
        <div
          class="css-zrbfnb"
        >
          <input
            class="chakra-input css-b2ycb2"
            placeholder="Search..."
            type="text"
          />
          <button
            aria-label="Search database"
            class="chakra-button css-1ljo2n9"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-bokek7"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="css-aozu9f"
      >
        <div
          class="css-ub1tjl"
        >
          <div
            class="css-7ccns3"
          >
            <div
              class="css-ervdjc"
            >
              <p
                class="chakra-text css-1ok93o9"
              >
                -25%
              </p>
            </div>
            <img
              alt="Lira Earrings"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/images/liraEarring.svg?w=640&q=75"
              srcset="/images/liraEarring.svg?w=384&q=75 1x, /images/liraEarring.svg?w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
            <div
              class="css-1xzknfv"
            >
              <h3
                class="chakra-heading css-7ex3dd"
              >
                <a
                  href="/products/1"
                >
                  Lira Earrings
                </a>
              </h3>
              <p
                class="chakra-text css-1ovs4to"
              >
                $ 20.00
              </p>
            </div>
          </div>
          <div
            class="css-7ccns3"
          >
            <img
              alt="Lira Earrings"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/images/liraEarring.svg?w=640&q=75"
              srcset="/images/liraEarring.svg?w=384&q=75 1x, /images/liraEarring.svg?w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
            <div
              class="css-1xzknfv"
            >
              <h3
                class="chakra-heading css-7ex3dd"
              >
                <a
                  href="/products/2"
                >
                  Lira Earrings
                </a>
              </h3>
              <p
                class="chakra-text css-1ovs4to"
              >
                $ 20.00
              </p>
            </div>
          </div>
          <div
            class="css-7ccns3"
          >
            <img
              alt="Kaede Hair Pin Set Of 3"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/images/liraEarring.svg?w=640&q=75"
              srcset="/images/liraEarring.svg?w=384&q=75 1x, /images/liraEarring.svg?w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
            <div
              class="css-1xzknfv"
            >
              <h3
                class="chakra-heading css-7ex3dd"
              >
                <a
                  href="/products/3"
                >
                  Kaede Hair Pin Set Of 3
                </a>
              </h3>
              <p
                class="chakra-text css-1ovs4to"
              >
                $ 20.00
              </p>
            </div>
          </div>
          <div
            class="css-7ccns3"
          >
            <img
              alt="Hair Pin Set of 3"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/images/liraEarring.svg?w=640&q=75"
              srcset="/images/liraEarring.svg?w=384&q=75 1x, /images/liraEarring.svg?w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
            <div
              class="css-1xzknfv"
            >
              <h3
                class="chakra-heading css-7ex3dd"
              >
                <a
                  href="/products/4"
                >
                  Hair Pin Set of 3
                </a>
              </h3>
              <p
                class="chakra-text css-1ovs4to"
              >
                $ 20.00
              </p>
            </div>
          </div>
          <div
            class="css-7ccns3"
          >
            <div
              class="css-ervdjc"
            >
              <p
                class="chakra-text css-1ok93o9"
              >
                Sold out
              </p>
            </div>
            <img
              alt="Plaine Necklace"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/images/liraEarring.svg?w=640&q=75"
              srcset="/images/liraEarring.svg?w=384&q=75 1x, /images/liraEarring.svg?w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
            <div
              class="css-1xzknfv"
            >
              <h3
                class="chakra-heading css-7ex3dd"
              >
                <a
                  href="/products/5"
                >
                  Plaine Necklace
                </a>
              </h3>
              <p
                class="chakra-text css-1ovs4to"
              >
                $ 20.00
              </p>
            </div>
          </div>
          <div
            class="css-7ccns3"
          >
            <img
              alt="Yuki Hair Pin Set of 3"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/images/liraEarring.svg?w=640&q=75"
              srcset="/images/liraEarring.svg?w=384&q=75 1x, /images/liraEarring.svg?w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
            <div
              class="css-1xzknfv"
            >
              <h3
                class="chakra-heading css-7ex3dd"
              >
                <a
                  href="/products/6"
                >
                  Yuki Hair Pin Set of 3
                </a>
              </h3>
              <p
                class="chakra-text css-1ovs4to"
              >
                $ 20.00
              </p>
            </div>
          </div>
        </div>
        <button
          class="chakra-button css-5p8da"
          data-loading=""
          disabled=""
          type="button"
        >
          <div
            class="chakra-button__spinner chakra-button__spinner--start css-1aamklj"
          >
            <div
              class="chakra-spinner css-1wpg9jd"
            >
              <span
                class="css-8b45rq"
              >
                Loading...
              </span>
            </div>
          </div>
          <span
            class="css-q7lffx"
          >
            Load More
          </span>
        </button>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
